public/src/pages/TeacherPage.tsx

```diff
@@ -154,6 +157,37 @@ export class TeacherPage extends ViewPage {
         return <div>404 Page not found</div>;
     }
 
+    public async groupresults(info: INavInfo<{ course: string }>): View {
+        const courseId = parseInt(info.params.course, 10);
+        const course = await this.courseMan.getCourse(courseId);
+        if (course) {
```

Comment:
These three lines appear in all functions in `TeacherPage.tsx`, except `editGroup`, which has a slight variation so should be handled separately for now at least. We can simply make a closure `courseFunc` that executes the code inside the `if (course)` block on the provided `courseParam: string`.
```typescript
    private async courseFunc(courseParam: string, fn: (course: ICourse) => View): View {
        const courseId = parseInt(courseParam, 10);
        const course = await this.courseMan.getCourse(courseId);
        if (course) {
            return fn(course)
        }
        return <div>404 Page not found</div>;
    }
```
And you would call it like this:
```typescript
        return this.courseFunc(info.params.course, async (course) => {
            // HERE GOES THE DIFFERENT FUNCTIONS; REPLACE LINES BELOW.
            const groups = await this.courseMan.getCourseGroups(course.id);
            ...
            return <CourseGroup ...;
        })
```
Note that we now would need to replace `courseId` with `course.id`.

Reason to do this: It helps keep functions related to parsing ints from strings in one place. Later we could probably make this closure function generic, so that it could also work on other types, such as `IUser` and `IGroup` if necessary.

_Note that I'm not a big fan of all these `async` and `await` annotations. It is not at all clear to me if they are needed everywhere. My understanding is that they are useful when interacting with the backend, but it feels like much of this stuff does not. I have not investigated._

Reply:
Pushed an example to the branch.